catalog/resource_lakehouse_monitor.go

```diff
@@ -0,0 +1,120 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const DefaultProvisionTimeout = 15 * time.Minute
+
+func WaitForMonitor(w *databricks.WorkspaceClient, ctx context.Context, monitorName string) error {
+	return retry.RetryContext(ctx, DefaultProvisionTimeout, func() *retry.RetryError {
+		endpoint, err := w.LakehouseMonitors.GetByFullName(ctx, monitorName)
```

Comment:
The SDK supports long-running operations automatically through the `x-databricks-wait` annotation. Will send you the link offline. This will let you use the following standard pattern for long-running operations:
```go
wait, err := w.LakehouseMonitors.Create(ctx, req)
if err != nil {
  return err
}
wait.Get()
```